src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,32 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction() {
+                return null;
```

Comment:
I think using constant for USE_FROM_SERVICE is not feasible here, because the return type of getServiceWriteTransaction is SProcService.WriteTransaction.

I prefer adding a parameter to getServiceWriteTransaction what is the sproc service write transaction mode.

Something like:
 public SProcService.WriteTransaction getServiceWriteTransaction(SprocService.WriteTransaction serviceTransaction)

So for USE_FROM_SERVICE we can simply return the input parameter.

If i could not understand you correctly, could you please explain it more by some code snippet?
Thanks